eZ/Publish/Core/Repository/Helper/DomainMapper.php

```diff
@@ -261,6 +261,21 @@ public function buildVersionInfoDomainObject(SPIVersionInfo $spiVersionInfo, arr
      */
     public function buildContentInfoDomainObject(SPIContentInfo $spiContentInfo)
     {
+        // Map SPI statuses to API
+        switch ($spiContentInfo->status) {
+            case SPIContentInfo::STATUS_TRASHED:
+                $status = ContentInfo::STATUS_TRASHED;
+                break;
+
+            case SPIContentInfo::STATUS_PUBLISHED:
+                $status = ContentInfo::STATUS_PUBLISHED;
+                break;
+
+            case SPIContentInfo::STATUS_DRAFT:
+            default:
+                $status = ContentInfo::STATUS_DRAFT;
```

Comment:
We can't throw exception here, not without potentially big BC break on a critical core feature like building ContentInfo domain object. It would make most parts of a system unusable.